simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -1862,9 +1838,7 @@ class basic_simd<std::int64_t, simd_abi::avx2_fixed_size<4>> {
   using abi_type   = simd_abi::avx2_fixed_size<4>;
   using mask_type  = basic_simd_mask<value_type, abi_type>;
 
-  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION static constexpr std::size_t size() {
-    return 4;
-  }
+  static constexpr std::integral_constant<simd_size_t, 4> size{};
```

Comment:
I hope this does not break something downstream, but if it does they were doing stuff that was brittle...

Reply:
It is experimental, so no guarantee things won't change -- but also we have to do it this way if we want to be as close to standard as possible